icu4c/source/common/uresbund.cpp

```diff
@@ -2632,7 +2632,8 @@ ures_getFunctionalEquivalent(char *result, int32_t resultCapacity,
 #endif
         if(U_FAILURE(subStatus)) {
             *status = subStatus;
-        } else if(subStatus == U_ZERO_ERROR) {
+        } else if(subStatus == U_ZERO_ERROR || subStatus == U_USING_DEFAULT_WARNING) {
```

Comment:
For reference: This looks similar to ICU-20431 (PR #377) and the fix in commit 290feb3d8f265ee40b1696b14071a7e48041f263 which I first proposed for it in January, but which then was rejected in the TC meeting and replaced with the fix in commit 092afb3d891a0a495e34de935d3382ec051db7d9 instead.

Reply:
Particularly, it's a regression in ICU 63. Do you know which commmit caused the regression?

Reply:
> How does commit #092afb3 fix this issue?

It does not, it fixes a different (but somewhat related) issue.

> Do you know which commmit caused the regression?

 Sorry, no.

Reply:
Actually, since your change was not accepted, I guess I suspect that an invalid locale "abc" should not fallback to the default locale. 

But I think it should at least fallback to the root locale. What do you think?

Reply:
https://github.com/unicode-org/icu/commit/b12a927c9365bb38831afbf76fdd0999f8f33deb
this commit caused the regression. 
Specifically, the new error check in DateTimePatternGenerator::getCalendarTypeToUse

Reply:
Thanks for tracking down the change @gvictor.

Here is a link to the line in the PR diff:
https://github.com/unicode-org/icu/commit/b12a927c9365bb38831afbf76fdd0999f8f33deb#diff-14f1d714f00a6f21e52eae73ad7fa1e4R771

The change added a check for failure after calling `ures_getFunctionalEquivalent`, as before there was no check at all.
If `ures_getFunctionalEquivalent` failed (imagine an out-of-memory failure) then the code would continue and would eventually return `U_ZERO_ERROR`.

I don't think hiding catastrophic errors like OOM is good practice, so I'm somewhat inclined to keep the check in place. Perhaps it could be weaked to allow `U_MISSING_RESOURCE_ERROR` though?